lib/lotus/cli_sub_commands/destroy.rb

```diff
@@ -0,0 +1,89 @@
+require 'lotus/routing/route'
+require 'lotus/commands/generate/action'
+
+module Lotus
+  class CliSubCommands
+    class Destroy < Thor
+      include Thor::Actions
+      namespace :destroy
+
+      desc 'action APPLICATION_NAME CONTROLLER_NAME#ACTION_NAME', 'destroy a lotus action'
+      long_desc <<-EOS
+        `lotus destroy action` will destroy an an action, view and template along with specs and a route.
+
+        For Application architecture the application name is 'app'. For Container architecture the default application is called 'web'.
+
+        > $ lotus destroy action app cars#index
+
+        > $ lotus destroy action other-app cars#index
+
+        > $ lotus destroy action web cars#create --method=post
+      EOS
+
+      method_option :method, desc: "The HTTP method used when the route was generated. Must be one of (#{Lotus::Routing::Route::VALID_HTTP_VERBS.join('/')})", default: Lotus::Commands::Generate::Action::DEFAULT_HTTP_METHOD
+      method_option :url, desc: 'Relative URL for action, will be used for the route', default: nil
+      method_option :template, desc: 'Extension used when the template was generated. Default is defined through your .lotusrc file.'
+
+      def actions(application_name, controller_and_action_name)
+        if options[:help]
+          invoke :help, ['action']
+        else
+          require 'lotus/commands/generate/action'
+
+          Lotus::Commands::Generate::Action.new(options, application_name, controller_and_action_name).destroy.start
+        end
+      end
+
+      desc 'migration NAME', 'destroy a migration'
+      long_desc <<-EOS
+      `lotus destroy migration` will destroy a migration file.
+
+      > $ lotus destroy migration create_books
```

Comment:
The most recent migration will be destroyed, but migrations with same naming is really desired?

The changes made in the migration generator also avoids generate migrations with same name, instead it will tell that a migration already exists and ask if can overwrite it. This is the same behavior of the rails migration generator.